Stripe/STPEphemeralKeyManager.m

```diff
@@ -64,18 +68,44 @@ - (void)handleWillForegroundNotification {
     // eager refreshses to once per hour.
     if (!self.currentKeyIsUnexpired && self.shouldPerformEagerRefresh) {
         self.lastEagerKeyRefresh = [NSDate date];
-        [self.keyProvider createCustomerKeyWithAPIVersion:self.apiVersion completion:^(NSDictionary *jsonResponse, __unused NSError *error) {
-            STPEphemeralKey *key = [STPEphemeralKey decodedObjectFromAPIResponse:jsonResponse];
-            if (key) {
-                self.customerKey = key;
-            }
+        [self getOrCreateKey:^(__unused STPEphemeralKey * _Nullable ephemeralKey, __unused NSError * _Nullable error) {
+            
         }];
     }
 }
 
-- (void)getCustomerKey:(STPEphemeralKeyCompletionBlock)completion {
+- (void)createKey {
+    STPJSONResponseCompletionBlock jsonCompletion = ^(NSDictionary *jsonResponse, NSError *error) {
+        STPEphemeralKey *key = [STPEphemeralKey decodedObjectFromAPIResponse:jsonResponse];
+        if (key) {
+            [self.createKeyPromise succeed:key];
+        } else {
+            // the API request failed
+            if (error) {
+                [self.createKeyPromise fail:error];
+            }
+            // the ephemeral key could not be decoded
+            else {
```

Comment:
style-nit: I prefer to have the `else` on the same line as the closing brace, `} else {`

Reply:
👍